dask/array/routines.py

```diff
@@ -780,6 +780,30 @@ def unique(ar, return_index=False, return_inverse=False, return_counts=False):
     return result
 
 
+def _isin_kernel(element, test_elements, assume_unique=False):
+    values = np.in1d(element.ravel(), test_elements,
+                     assume_unique=assume_unique)
+    return values.reshape(element.shape + (1,) * test_elements.ndim)
+
+
+@wraps(getattr(np, 'isin', None))
+def isin(element, test_elements, assume_unique=False, invert=False):
+    element = asarray(element)
+    test_elements = asarray(test_elements)
```

Comment:
On a different point, it looks like NumPy's `isin` flattens the `test_elements` array first. This might simplify some of the chunk handling below and in the kernel function. Thoughts on doing this here?

Reply:
I don't see a major advantage to doing this. reshaping might be slightly expensive if lots of chunks need to move around.